### PR TITLE
fix: [CDS-50960]: added test coverage for artifactdigest

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger cypress-cd

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx [hunk 1]
```diff
@@ -58,6 +60,13 @@ export function NoDigestResults({
   )
 }
 
+const getItems = (isFetching: boolean, label: string, items: SelectOption[]): SelectOption[] => {
+  if (isFetching) {
+    return [{ label: `Loading ${label}...`, value: `Loading ${label}...` }]
+  }
+  return defaultTo(items, [])
```
Comment (devinder-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx:67]: why we need `defaultTo(items, [])` ? when we already have `items: SelectOption[]` which is `non-optional` ?

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx [hunk 2]
```diff
@@ -80,8 +89,8 @@ function ArtifactDigestField({
   const { getString } = useStrings()
   const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
   const loadingPlaceholderText = getString('pipeline.artifactsSelection.loadingDigest')
-  const digestDefaultValue = [{ label: loadingPlaceholderText, value: loadingPlaceholderText }]
-  const [digestItems, setDigestItems] = React.useState<any>(digestDefaultValue)
+  // const digestDefaultValue = [{ label: loadingPlaceholderText, value: loadingPlaceholderText }]
+  // const [digestItems, setDigestItems] = React.useState<any>(digestDefaultValue)
```
Comment (devinder-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx:93]: We can remove commented code

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx [hunk 3]
```diff
@@ -101,35 +110,34 @@ function ArtifactDigestField({
...
-      /* istanbul ignore next */
-      const options = []
-      if (data && data?.data && data?.data?.metadata && data?.data?.metadata?.SHA) {
-        options.push({ label: data.data.metadata.SHA, value: data.data.metadata.SHA })
+  const digestItems: SelectOption[] = React.useMemo(() => {
+    const options = []
+    if (loading) {
+      options.push({ label: loadingPlaceholderText, value: loadingPlaceholderText })
+    } else {
+      if (data && get(data, 'data', {}) && get(data, 'data.metadata', {}) && get(data, 'data.metadata.SHA', '')) {
```
Comment (devinder-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx:118]: can be written as `if(get(data, 'data.metadata.SHA', ''))` ?

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx [hunk 4]
```diff
@@ -101,35 +110,34 @@ function ArtifactDigestField({
...
+        /* istanbul ignore next */
+        options.push({ label: get(data, 'data.metadata.SHA', ''), value: get(data, 'data.metadata.SHA', '') })
       }
-      if (data && data?.data && data?.data?.metadata && data?.data?.metadata?.SHAV2) {
-        options.push({ label: data.data.metadata.SHAV2, value: data.data.metadata.SHAV2 })
+      if (data && get(data, 'data', {}) && get(data, 'data.metadata', {}) && get(data, 'data.metadata.SHAV2', '')) {
```
Comment (devinder-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx:122]: same as above

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx (hunk 3 above)
Comment (sahithikolichala) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx:118]: ya will change to that

**Review by satyam-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx [hunk 5]
```diff
@@ -58,6 +60,13 @@ export function NoDigestResults({
   )
 }
 
+const getItems = (isFetching: boolean, label: string, items: SelectOption[]): SelectOption[] => {
+  if (isFetching) {
+    return [{ label: `Loading ${label}...`, value: `Loading ${label}...` }]
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx:65]: These values should not be hard coded here. These should comes from yaml.

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx (hunk 5 above)
Comment (sahithikolichala) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx:65]: ok will change this

Comment (sahithikolichala) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactDigestField.tsx]: fixed this https://github.com/harness/harness-core-ui/pull/13691/files#diff-6757bde8318ac524dfd8a0ce2a2a72ae4c419b0b68fe6364bfba5370743a758cR65

as the loading string is coming from https://github.com/harness/harness-core-ui/pull/13691/files#diff-6757bde8318ac524dfd8a0ce2a2a72ae4c419b0b68fe6364bfba5370743a758cR65

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger cypress-cd

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/__tests/ArtifactDigestField.test.tsx [hunk 6]
```diff
@@ -87,4 +99,64 @@ describe('ArtifactDigest tests', () => {
...
+    })
+    const { container } = render(
+      <TestWrapper>
+        <ArtifactDigestField {...props} />
+      </TestWrapper>
+    )
+
+    const portalDivs = document.getElementsByClassName('bp3-portal')
+    expect(portalDivs.length).toBe(0)
+    expect(container).toMatchSnapshot()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/__tests/ArtifactDigestField.test.tsx:119]: We can avoid snapshot testing.

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/__tests/ArtifactDigestField.test.tsx [hunk 7]
```diff
@@ -87,4 +99,64 @@ describe('ArtifactDigest tests', () => {
...
+    const packageDropdwnBtn = container.querySelectorAll('[data-icon="chevron-down"]')[0]
+    userEvent.click(packageDropdwnBtn!)
+
+    const dropdownPortalDiv = portalDivs[0]
+    const selectListMenu = dropdownPortalDiv.querySelector('.bp3-menu')
+
+    const loadingBucketsOption = await findByText(
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/__tests/ArtifactDigestField.test.tsx:126]: variable name could be `loadingDigestOption`

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/__tests/ArtifactDigestField.test.tsx [hunk 8]
```diff
@@ -87,4 +99,64 @@ describe('ArtifactDigest tests', () => {
...
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/__tests/ArtifactDigestField.test.tsx:156]: variable...

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Pull request merged**